core/views.py

```diff
@@ -46,7 +46,22 @@ class UnitGroupListView(LoginRequiredMixin, ListView[Unit]):
     template_name = "core/unit_list.html"
 
     def get_queryset(self):
-        queryset = Unit.objects.filter(group__hidden=False)
+        assert isinstance(user := self.request.user, User)
+        student = Student.objects.filter(user=user, semester__active=True).first()
```

Comment:
I think this has the effect that, e.g. once summer starts, the secret units disappear, and only reappear if the student re-enrolls in OTIS.

I'd suggest that we actually take the `Max` of `last_level_seen` across all the students? (Of course then we should handle the case of empty `Max`.)

Reply:
Something like the following (please test and make sure it actually works :P )

```python
level : int | None = Student.objects.filter(user=user).aggregate(
    level=Max("last_level_seen")
)["level"]
```